### PR TITLE
Fixup ChibiOS header inclusion search ordering.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

Added Nucleo-STM32F446 onekey.

In the process, noticed that overrides at board level for board.h weren't working, due to wrong search path ordering.
Docs and template already specify using `<board.h>`, but a lot of existing files already use `"board.h"`.

`chconf.h`, `halconf.h`, and `mcuconf.h` all similarly afflicted... and fixed in the PR.

Adds a new lint check against all four files as well, to try to guard against future problematic commits.

I'm expecting a failure on the auto-merge to `develop`, due to collisions and file moves.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
Made it a draft while I sort out the rest of the onekey keymaps.

File: lib/python/qmk/cli/lint.py [hunk 1]
```diff
@@ -153,6 +171,12 @@ def keyboard_check(kb):
             cli.log.error(f'{kb}: The file "{file}" does not have a license header!')
             ok = False
 
+        if file.name in CHIBIOS_CONF_CHECKS:
+            check_error = _chibios_conf_includenext_check(file)
+            if check_error is not None:
+                cli.log.error(check_error)
```
Comment (zvecr) [inline lib/python/qmk/cli/lint.py:177]:
```suggestion
                cli.log.error(f'{kb}: {check_error')
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: lib/python/qmk/cli/lint.py [hunk 2]
```diff
@@ -64,6 +66,16 @@ def _handle_json_errors(kb, info):
...
 
 
+def _chibios_conf_includenext_check(path):
+    """Check the ChibiOS conf.h for the correct inclusion of the next conf.h
+    """
+    target = Path(path)
+    for i, line in enumerate(target.open()):
+        if f'#include_next "{target.name}"' in line:
+            return f'Found `#include_next "{target.name}"` on line {i} of {path}, should be `#include_next <{target.name}>` (use angle brackets, not quotes)'
+    return None
```
Comment (zvecr) [inline lib/python/qmk/cli/lint.py:76]: Isnt the argument always a `Path`?
```suggestion
def _chibios_conf_includenext_check(target):
    """Check the ChibiOS conf.h for the correct inclusion of the next conf.h
    """
    for i, line in enumerate(target.open()):
        if f'#include_next "{target.name}"' in line:
            return f'Found `#include_next "{target.name}"` on line {i} of {target}, should be `#include_next <{target.name}>` (use angle brackets, not quotes)'
    return None
```

Comment (tzarc) [reply lib/python/qmk/cli/lint.py]: Yeah, it wasn't on first pass as I wasn't using `_get_code_files()`.

**Review by zvecr — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**